netinterface.py

```diff
@@ -25,7 +25,7 @@ class NetInterface:
 
     def __init__(self, mode, hwaddr, ipaddr=None, netmask=None, gateway=None,
                  dns=None, domain=None, vlan=None):
-        assert mode is None or mode == self.Static or mode == self.DHCP
+        assert mode in [None, self.Static, self.DHCP, self.Autoconf]
```

Comment:
Distinct from the addition of a new value, there is a similar assert in `addIPv6` which would benefit from the same refactoring.  Would be worth having a separate preparatory commit to improve all of them.